tools/elfutils: use modern libelf on linux hosts

curtdept opened the pull request:
This is based on prior work @ https://github.com/openwrt/openwrt/pull/3855

This change is required to build linux kernels 5.8+ as the existing, quite old, libelf does not support gelf_getsymshndx. See https://github.com/torvalds/linux/blob/4e909124f8ed54b13e07e42273c3452b7deb5a0b/tools/objtool/elf.c#L378

```Signed-off-by: Curtis Deptuck <curtdept@me.com>```

adschm commented on tools/elfutils/Makefile:
```diff
@@ -0,0 +1,42 @@
+	$(call Host/Clean/Default)
+endef
+
+$(eval $(call HostBuild))
```
Missing newline at EOF here and for the next one.

curtdept replied:
Thanks will fix

curtdept replied:
Fixed :)

curtdept committed:
tools/elfutils: use modern libelf on linux hosts

This is based on prior work @ https://github.com/openwrt/openwrt/pull/3855

This change is required to build linux kernels 5.8+ as the existing, quite old, libelf does not support gelf_getsymshndx. See https://github.com/torvalds/linux/blob/4e909124f8ed54b13e07e42273c3452b7deb5a0b/tools/objtool/elf.c#L378

Signed-off-by: Curtis Deptuck <curtdept@me.com>

guidosarducci commented:
@adschm The issue of outdated and problematic elf libs has already been discussed and raised, and is resolved with the first 3 commits of https://github.com/openwrt/openwrt/pull/3855. Needless reproduction of PRs only adds confusion and further delay to any sort of review and resolution.

@curtdept Rather than linking an active PR and than duplicating it, I think a better approach would be to simply confirm those commits worked for you, and then encourage a dev to pull those out into their staging tree, or cherry-pick, or take the whole PR. In my OP I already mentioned several devs with some relevant history, so pinging them would be a good place to start.

curtdept commented:
Happy to close if that's what we feel is necessary. My intent was to reduce your PR to my discrete parts that allowed 5.10 x86 to get over the finish line.

curtdept commented:
Closing as requested
